core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -70,4 +70,45 @@ public static String propertyAsString(Map<String, String> properties,
     }
     return defaultValue;
   }
+
+  @FunctionalInterface
+  private interface PropertyResolver<A, B, C, R> {
+    R apply(A argA, B argB, C argC);
```

Comment:
Looks like this is always `Map<String, String> properties, String property, R defaultVal`. Can we just parameterize this by `R`? Then we can also use sensible names, like `properties`, `property`, and `defaultVal`